core/primitives/Cargo.toml

```diff
@@ -48,6 +49,8 @@ near-fmt.workspace = true
 near-primitives-core.workspace = true
 near-rpc-error-macro.workspace = true
 near-parameters.workspace = true
+near-structs-checker.workspace = true
+near-structs-checker-core = { workspace = true, optional = true }
```

Comment:
Rather than having the crates depend on 3 crates at once export the macro and the necessary supporting crates from within it.